daml-lf/spec/daml-lf-1.rst

```diff
@@ -1955,187 +1975,326 @@ exact output.
+
+       eₕ  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpConsErr1
+      'Cons' @τ eₕ eₜ  ⇓  Err t
 
-       eₕ ‖ E₀  ⇓  Ok vₕ ‖ E₁
-       eₜ ‖ E₁  ⇓  Ok vₜ ‖ E₂
+       eₕ  ⇓  Ok vₕ
+       eₜ  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpConsErr2
+      'Cons' @τ eₕ eₜ  ⇓  Err t
+
+       eₕ  ⇓  Ok vₕ
+       eₜ  ⇓  Ok vₜ
     —————————————————————————————————————————————————————————————————————— EvExpCons
-      'Cons' @τ eₕ eₜ ‖ E₀  ⇓  Ok ('Cons' @τ vₕ vₜ) ‖ E₂
+      'Cons' @τ eₕ eₜ  ⇓  Ok ('Cons' @τ vₕ vₜ)
+
+       e  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpSomeErr
+      'Some' @τ e  ⇓  Err t
 
-       e ‖ E₀  ⇓  Ok v ‖ E₁
+       e  ⇓  Ok v
     —————————————————————————————————————————————————————————————————————— EvExpSome
-      'Some' @τ e ‖ E₀  ⇓  Ok ('Some' @τ v) ‖ E₂
+      'Some' @τ e  ⇓  Ok ('Some' @τ v)
+
+      𝕋(F) = ∀ (α₁: ⋆). … ∀ (αₘ: ⋆). σ₁ → … → σₙ → σ
+      e₁  ⇓  Ok v₁
+        ⋮
+      eᵢ  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpBuiltinErr
+      F @τ₁ … @τₘ e₁ … eₙ  ⇓  Err t
 
       𝕋(F) = ∀ (α₁: ⋆). … ∀ (αₘ: ⋆). σ₁ → … → σₙ → σ
-      e₁ ‖ E₀  ⇓  Ok v₁ ‖ E₁
+      e₁  ⇓  Ok v₁
         ⋮
-      eₙ ‖ Eₙ₋₁  ⇓  Ok vₙ ‖ Eₙ
+      eₙ  ⇓  Ok vₙ
     —————————————————————————————————————————————————————————————————————— EvExpBuiltin
-      F @τ₁ … @τₘ eᵢ … eₙ ‖ E₀  ⇓  𝕆(F @τ₁ … @τₘ v₁ … vₙ) ‖ Eₙ
+      F @τ₁ … @τₘ e₁ … eₙ  ⇓  𝕆(F @τ₁ … @τₘ v₁ … vₙ)
 
-      'val' W : τ ↦ e  ∈ 〚Ξ〛Mod      Mod:W ↦ … ∉ Eₒ
-      e ‖ E₀  ⇓  Ok v ‖ E₁
-    —————————————————————————————————————————————————————————————————————— EvExpNonCachedVal
-      Mod:W ‖ E₀  ⇓  Ok v ‖ Mod:W ↦ v · E₁
+      'val' W : τ ↦ e  ∈ 〚Ξ〛Mod
+      e  ⇓  r
+    —————————————————————————————————————————————————————————————————————— EvExpVal
+      Mod:W  ⇓  r
 
-      Mod:W ↦ v ∈ E₀
-    —————————————————————————————————————————————————————————————————————— EvExpCachedVal
-      Mod:W ‖ E₀  ⇓  Ok v ‖ E₀
+      e₁  ⇓  Ok v₁
+        ⋮
+      eᵢ  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpRecConErr
+      Mod:T @τ₁ … @τₘ {f₁ = e₁, …, fₙ = eₙ}
+        ⇓
+      Err t
 
-      e₁ ‖ E₀  ⇓  Ok v₁ ‖ E₁
+      e₁  ⇓  Ok v₁
         ⋮
-      eₙ ‖ Eₙ₋₁  ⇓  Ok vₙ ‖ Eₙ
+      eₙ  ⇓  Ok vₙ
     —————————————————————————————————————————————————————————————————————— EvExpRecCon
-      Mod:T @τ₁ … @τₘ {f₁ = e₁, …, fₙ = eₙ} ‖ E₀
+      Mod:T @τ₁ … @τₘ {f₁ = e₁, …, fₙ = eₙ}
         ⇓
-      Ok (Mod:T @τ₁ … @τₘ {f₁ = v₁, …, fₙ = ₙ}) ‖ Eₙ
+      Ok (Mod:T @τ₁ … @τₘ {f₁ = v₁, …, fₙ = ₙ})
 
-      e ‖ E₀  ⇓  Ok (Mod:T @τ₁ … @τₘ {f₁= v₁, …, fᵢ= vᵢ, …, fₙ= vₙ}) ‖ E₁
+      e  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpRecProjErr
+      Mod:T @τ₁ … @τₘ {fᵢ} e  ⇓  Err t
```

Comment:
Our syntax for record projections looks really weird. 🤔

Reply:
@hurryabit please propose a syntax that we can easily add to the daml-lf parser.